rest-client/impl/impl-base/src/main/java/org/jboss/arquillian/extension/rest/client/BaseRestEnricher.java

```diff
@@ -89,25 +86,25 @@ private void addHeaders(Map<String, String> headersMap, Header annotation)
         }
     }
 
-    protected boolean allInSameContext(List<Servlet> servlets)
-    {
-        Set<String> context = new HashSet<String>();
-        for (Servlet servlet : servlets) {
-            context.add(servlet.getContextRoot());
-        }
-        return context.size() == 1;
-    }
-
     protected abstract Object enrichByType(Class<?> clazz, Method method, ArquillianResteasyResource annotation, Consumes consumes, Produces produces);
 
-    // Currently no way to share @ArquillianResource URL (URLResourceProvider) logic internally, copied logic
+    // Currently no way to request value of URI, need to manually invoke the lookup
     protected URI getBaseURL()
     {
-        HTTPContext context = metaDataInst.get().getContext(HTTPContext.class);
-        if (allInSameContext(context.getServlets())) {
-            return context.getServlets().get(0).getBaseURI();
+        Collection<ResourceProvider> resourceProviders = loader.get().all(ResourceProvider.class);
+        for(ResourceProvider resourceProvider: resourceProviders)
+        {
+            if(resourceProvider.canProvide(URI.class))
```

Comment:
@johnament would that help?  https://github.com/arquillian/arquillian-core/pull/118 -> https://issues.jboss.org/browse/ARQ-2053

In fact it was already available long before, as you can read from the discussion on the PR https://github.com/arquillian/arquillian-core/pull/118#issuecomment-268784921, but well hidden :)